### PR TITLE
Remove errors from Typescript usage

### PR DESCRIPTION
**Pull request by barrymay**
- Disable non-critical eslint rules
- Add react rules for eslint
- Update ts-eslint parser to fix latest bugs
- Minor changes to fix bugs and avoid variable check errors

### CODE REVIEW

**Comment by bluebill1049** [comment]
Thanks, @barrymay heaps for the PR, I will take a look this soon. ❤️ 👏

File: package.json [hunk 1]
```diff
@@ -57,5 +57,8 @@
   "peerDependencies": {
     "react": "^16.8.0",
     "react-dom": "^16.8.0"
+  },
+  "dependencies": {
+    "eslint-plugin-react": "^7.13.0"
```
Comment (bluebill1049) [inline package.json]: I think this should be dev dependency

**Comment by bluebill1049** [comment]
only one minor comment :) looks good

File: package.json (hunk 1 above)
Comment (barry4dev) [reply package.json]: That’s fair - will fix

**Comment by bluebill1049** [comment]
thank you :) merging in

**Pull request merged**